extension/src/sbm/sbm-generation.ts

```diff
@@ -0,0 +1,540 @@
+ *
+ * http://rtsys.informatik.uni-kiel.de/kieler
+ *
+ * Copyright 2023 by
+ * + Kiel University
+ *   + Department of Computer Science
+ *     + Real-Time and Embedded Systems Group
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+import * as vscode from 'vscode';
+import { EMPTY_STATE_NAME, Enum, LTLFormula, State, Transition, UCA_TYPE, Variable } from "./utils";
+import { createSCChartText } from "./scchart-creation";
+import { createFile } from '../utils';
+
+/**
+ * Creates a safe behavioral model for each controller.
+ * @param controlActionsMap The control actions for each controller.
+ * @param formulaMap The ltl formulas for each controller.
+ */
+export function createSBMs(controlActionsMap: Record<string, string[]>, formulaMap: Record<string, LTLFormula[]>): void {
+    Object.keys(controlActionsMap).forEach(controller => createControllerSBM(controller, controlActionsMap[controller], formulaMap[controller]));
+}
+
+/**
+ * Creates a safe behavioral model with the given components.
+ * @param controllerName The name of the controller to model.
+ * @param controlActions The control actions of the controller.
+ * @param ltlFormulas The ltl formulas for the controller.
+ * @returns 
+ */
+async function createControllerSBM(controllerName: string, controlActions: string[], ltlFormulas: LTLFormula[]): Promise<void> {
+    // Ask the user where to save the sbm
+    const currentFolder = vscode.workspace.workspaceFolders
+        ? vscode.workspace.workspaceFolders[0].uri.fsPath
+        : undefined;
+    const uri = await vscode.window.showSaveDialog({
+        filters: { SCChart: ['sctx'] },
+        // TODO: not possible with current vscode version
+        // title: 'Save SBM to...',
+        defaultUri: currentFolder ? vscode.Uri.file(`${currentFolder}/sbm.sctx`) : undefined,
+    });
+    if (uri === undefined) {
+        // The user did not pick any file to save to.
+        return;
+    }
+
+    // group formulas and create states for control actions
+    const formulaMap = groupFormulasByActionAndType(ltlFormulas);
+    let states = createStatesForActions(controlActions);
+    // add transitions to the states
+    // when adding transition for continous UCA types new states may be added
+    states = addTransitions(formulaMap, states);
+    states = getReachableStates(states);
+
+    // sort the transitions so that the ones to the empty state are always the last transitions
+    /* Must be done because the transitions to the empty state should only ensure that the state is left (for types providing and applied-too-long). 
+        When another transition is possible, it should be taken. When not using priorities, the triggers must be adjusted to be exclusive */
+    states.forEach(state => {
+        state.transitions.sort((a, _) => {
```

Comment:
the linter is throwing a warning that _ is not used, maybe doublecheck the linter settings